dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/merge/ModelEquality.java

```diff
@@ -49,14 +49,24 @@ private static boolean equalsImpl(Object a, Object b) {
             return true;
         }
 
+        /*
+         * Take https://github.com/smallrye/smallrye-open-api/blob/3.13.0/core/src/main/java/io/smallrye/openapi/api/models/ExtensibleImpl.java
+         * extensions is null. If you call addExtension() it will become an map with an item.
+         * if you then call remove, it will remove the item from the list and not null it.
+         * In short new ExtensibleImpl().addExtension(x).removeExtension(x) is not equals() new ExtensibleImpl().
+         *
+         * So to work around that, we treat null and empty maps as equivalent.
+         *
+         * This may need to be extended to cover lists
+         */
         if (a == null) {
-            if (b == null) {
+            if (nullOrEmptyMap(b)) {
                 return true;
             } else {
                 return false;
             }
         } else if (b == null) {
-            return false;
+            return nullOrEmptyMap(a);
         }
```

Comment:
I think this is ok, but it's close. The [Security Requirement Object][sro] cares about a map being empty vs not being present in an array, but that doesn't quite hit this condition and I'm not aware of other cases.

[sro]: https://spec.openapis.org/oas/v3.1.0.html#security-requirement-object

Reply:
I added this because of one the FAT tests failed, I'd have to take this out to remind myself which on but from memory this lack of equality was causing us to output something like "merged document" when the fat test expected "Generated API".

Let me know if you'd like us to do a deep dive into this?